staging/src/k8s.io/kubectl/pkg/scale/scale.go

```diff
@@ -128,8 +124,8 @@ func (s *genericScaler) ScaleSimple(namespace, name string, preconditions *Scale
 		}
 	}
 
-	scale.Spec.Replicas = int32(newSize)
-	updatedScale, err := s.scaleNamespacer.Scales(namespace).Update(gr, scale)
+	patch := []byte(fmt.Sprintf(`{"spec":{"replicas":%d}}`, newSize))
+	updatedScale, err := s.scaleNamespacer.Scales(namespace).Patch(gvr, name, types.MergePatchType, patch)
```

Comment:
This is breaking the precondition though, you would have to send the "resourceVersion" in the patch if that's what you want, otherwise there might be a race condition. Or you could use update on preconditions, and do the patch otherwise.

Reply:
@apelisse That's a good point! I have switched to updating scale subresource on preconditions and patching otherwise, PTAL.